index.js

```diff
@@ -0,0 +1,88 @@
+const express = require('express');
+const bcrypt = require('bcryptjs');
+const db = require('./data/dbConfig.js');
+const session = require('express-session');
+const server = express();
+
+const PORT = 4200;
+
+server.use(express.json());
+
+server.use(session({
+    name: 'sessionTest',
+    secret: 'SECRET',
+    cookie: {
+      maxAge: 1 * 24 * 60 * 60 * 1000,
+    },
+    httpOnly: true,
+    resave: false,
+    saveUninitialized: false,
+}));
+
+function protect(req, res, next){
+    if (req.path.includes('/restricted/') || req.path == '/api/users') {
+        if (req.session && req.session.userId) {
+            next();
+        } else {
+            res.status(401).json({ message: 'go away' })
+        }
+        } else {
+        next();
+        }
+};
+
+server.use(protect);
+
+server.get('/', (req, res) => {
+    res.send('its alive!');
+});
+
+server.get('/api/users', (req, res) => {
+    db('users')
+    .select('id', 'username')
+    .then(users => {
+        res.json(users);
+    })
+    .catch(err => res.json(err));
+});
+
+server.post('/api/register', (req, res) => {
+    const creds = req.body;
+    db('users')
+        .where({ username: creds.username })
+        .first()
+        .then(user => {
+            if (user) {
```

Comment:
This if statement on line 55 may not be necessary because in your migrations you already specified that a username has to be unique, which means the db will not let you create another user automatically with the same username